core/tests/test_utils.py

```diff
@@ -3232,12 +3208,17 @@ def traverse_schema_and_assign_content_ids(
                 for x in value:
                     traverse_schema_and_assign_content_ids(
                         x, schema['items'], ca_name)
+            # We exclude this elif block from coverage because no interaction
```

Comment:
I think it's fine to actually just do an `else: raise Exception` here, because we won't introduce an unstructured dict type in the future now that we're typing everything fully.

Reply:
done